lib/matplotlib/tests/test_ticker.py

```diff
@@ -1594,6 +1594,79 @@ def test_engformatter_usetex_useMathText():
         assert x_tick_label_text == ['$0$', '$500$', '$1$ k']
 
 
+@pytest.mark.parametrize(
+    'oom_center, oom_noise, oom_center_desired, oom_noise_desired', [
+        (11, 1, 9, 0),
+        (13, 7, 12, 6),
+        (1, -2, 0, -3),
+        (3, -2, 3, -3),
+        (5, -3, 3, -3),
+        (2, -3, 0, -3),
+        # The following sets of parameters demonstrates that when oom_center-1
+        # and oom_noise-2 equal a standard 3*N oom, we get that
+        # oom_noise_desired < oom_noise
+        (10, 2, 9, 3),
+        (1, -7, 0, -6),
+        (2, -4, 0, -3),
+        (1, -4, 0, -3),
+        # Tests where oom_center <= oom_noise, those are probably covered by the
+        # part where formatter.offset != 0
+        (4,  4, 0, 3),
+        (1,  4, 0, 3),
+        (1,  3, 0, 3),
+        (1,  2, 0, 0),
+        (1,  1, 0, 0),
+    ]
+)
+def test_engformatter_offset_oom(
+    oom_center,
+    oom_noise,
+    oom_center_desired,
+    oom_noise_desired
+):
+    UNIT = "eV"
+    # Doesn't really matter here, but should be of order of magnitude ~= 1
+    r = range(-5, 7)
+    fig, ax = plt.subplots()
+    # Use some random ugly number
+    data_offset = 2.7149*10**oom_center
+    ydata = data_offset + np.array(r, dtype=float)*10**oom_noise
+    ax.plot(ydata)
+    formatter = mticker.EngFormatter(useOffset=True, unit=UNIT)
+    # So that offset strings will always have the same size
+    formatter.ENG_PREFIXES[0] = "_"
+    ax.yaxis.set_major_formatter(formatter)
+    fig.canvas.draw()
+    offsetGot = formatter.get_offset()
+    ticksGot = [labl.get_text() for labl in ax.get_yticklabels()]
```

Comment:
Though `ticker.py` is not great at it, please stick to PEP8 variable naming here.
```suggestion
    offset_got = formatter.get_offset()
    ticks_got = [labl.get_text() for labl in ax.get_yticklabels()]
```
(though I would just drop the `_got` entirely as there is no other version.)

Reply:
> (though I would just drop the `_got` entirely as there is no other version.)

That's another thing that helped me debug things with `pytest --showlocals`...

Reply:
> > (though I would just drop the `_got` entirely as there is no other version.)
> 
> That's another thing that helped me debug things with `pytest --showlocals`...

So I am marking this review thread as resolved, as I got the impression that the `--showlocals` dilemma is resolved.